src/types/any.rs

```diff
@@ -124,6 +124,55 @@ impl PyAny {
         }
     }
 
+    /// Retrieve an attribute value, skipping the instance dictionary during the lookup but still
+    /// binding the object to the instance.
+    ///
+    /// This is useful when trying to resolve Python's "magic" methods like `__getitem__`, which
+    /// are looked up starting from the type object.  This returns an `Option` as it is not
+    /// typically a direct error for the special lookup to fail, as magic methods are optional in
+    /// many situations in which they might be called.
+    ///
+    /// To avoid repeated temporary allocations of Python strings, the [`intern!`] macro can be used
+    /// to intern `attr_name`.
+    #[allow(dead_code)] // Currently only used with num-complex+abi3, so dead without that.
+    pub(crate) fn lookup_special<N>(&self, attr_name: N) -> PyResult<Option<&PyAny>>
+    where
+        N: IntoPy<Py<PyString>>,
+    {
+        let py = self.py();
+        let self_type = self.get_type();
+        let attr = if let Ok(attr) = self_type.getattr(attr_name) {
+            attr
+        } else {
+            return Ok(None);
+        };
```

Comment:
let-else is going to be so nice once PyO3 gets to MSRV of 1.65... but that's going to be a long time yet! (We're just about to do a bump to 1.56 I think.)

Reply:
How did this get through the full CI though? Are we missing a feature configuration for the MSRV job?

Reply:
Yes, right. In such cases, personally, I actually prefer `match` as it gives "tigher" formatting even though it does "stutter" a bit:

```rust
let attr = match self_type.getattr(attr_name) {
  Ok(attr) => attr,
  Err(_err) => return Ok(None),
};
```